src/palace/manager/api/admin/controller/integration_settings.py

```diff
@@ -381,6 +385,12 @@ def process_deleted_libraries(
         for library_integration in removed:
             self._db.delete(library_integration)
 
+        if removed:
+            # ensure that all loans and holds related
```

Comment:
@jonathangreen : These task calls are breaking the tests.  Two questions here:  1) Do you agree that these should go here.  2) If so,  is there something special I need to do to patch these methods when testing in the flask context?  For example: in [this collection settings controller test](https://github.com/ThePalaceProject/circulation/blob/main/tests/manager/api/admin/controller/test_collections.py#L696) I tread using a `with patch("palace.manager.celery.tasks.reaper.reap_unassociated_loans") as reap_loans:` context manager, but when I actually debug the test I notice that the reap_unassociated_loans function is not being mocked.

Reply:
1. This shouldn't go here. This class is a super-class for a lot of our settings controllers. It is not just for the collection settings. This method is called by all of them. That would mean we would trigger these celery tasks anytime a library is removed from any integration. I don't think we need to do that for all the different integration types like patron auth, or metadata, marc exporter, etc. You could either remove it entirely here, or target it at the specific controller you want.
2. The documentation for `unittest.mock` has a good section on where to patch https://docs.python.org/3/library/unittest.mock.html#where-to-patch that should help you.

Reply:
Thanks for that reference.  I will move it to the right controller.